### ISSUE TITLE
Dependency `graph`/`tree` command

### ISSUE THREAD

**Issue by johanfylling** [johanfylling/opa-dependency-manager#7]
Add a `tree` command for listing dependencies, and their transitive "children", as a tree graph.

**Comment by widal001** [comment]
@johanfylling Really excited about this project! And this looks like a fun issue to tackle if you're open to contributions. 

I had a couple of questions about the assumptions underlying this issue:

- Would this command assume the existence of a `.project` file in each dependency? If not, is there another strategy you were anticipating for retrieving the sub-dependencies?
- Should the interface for the command and the output be modeled after [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html#cargo-tree1)?
- Drawing inspiration from `cargo tree` I was thinking we could focus on the following flags in the v0. Does that seem reasonable to you?
  - `--prune`
  - `--depth`
  - `--no-dedupe`

**Comment by johanfylling** [comment]
Yes, `cargo tree` looks very much like what is envisioned for `odm tree` 👍.
I think the very first iteration of this can do without a lot of bells and whistles; that said, the suggested flags look very reasonable, especially `--no-dedupe`.

An ODM dependency can be any local folder or git repo that contains Rego code or json/yaml data, and an `opa.project` file is only necessary if there are transitive dependencies. So any dependency source without a project file is simply a "leaf" dependency, that has no further branches in the dependency tree. There are no plans to support disjoint dependency management, where the dependency structure is maintained separately from the source.
